core/src/main/java/org/ehcache/core/internal/util/CollectionUtil.java

```diff
@@ -50,4 +52,78 @@ public static int findBestCollectionSize(Iterable<?> iterable, int bestBet) {
   public static <K, V> Map.Entry<K, V> entry(K key, V value) {
     return new AbstractMap.SimpleImmutableEntry<>(key, value);
   }
+
+  /**
+   * Copy each map entry to a new map but check that each key and value isn't null. Throw
+   * a {@code NullPointerException} if it's the case.
+   *
+   * @param entries entries to copy
+   * @param <K> type of key
+   * @param <V> type of value
+   * @return cloned map
+   * @throws NullPointerException if a key or value is null
+   */
+  public static <K, V> Map<K, V> copyMapButFailOnNull(Map<? extends K, ? extends V> entries) {
```

Comment:
```java
public static <K, V> Map<K, V> nullSafeCopy(Map<? extends K, ? extends V> entries) {
  return entries.entrySet().stream().collect(toMap(
    e -> requireNonNull(e.getKey()),
    e -> requireNonNull(e.getValue())
  ));
}
```

Reply:
I don't think it's null safe. I think it doesn't like null. I've avoided the stream on purpose, I don't want the related object instantiation. Not using `requireNonNull` was also on purpose.